File: Odontio.Domain/Entities/Treatment.cs
namespace Odontio.Domain.Entities;

public class Treatment
{
    public long Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public decimal? Cost { get; set; }

    public long WorkspaceId { get; set; }
    public Workspace Workspace { get; set; } = null!;

    public long CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public ICollection<PatientTreatment> PatientTreatments { get; set; } = new List<PatientTreatment>();
}

File: Odontio.WebApp/src/types/treatment.ts
export type TreatmentDto = {
  id: number | null;
  name: string | null;
  description: string | null;
  cost: number | null;
  categoryName: string | null;
};

export type UpsertTreatmentDto = {
  name: string | null;
  description: string | null;
  cost: number | null;
  categoryId: number | null;
};

export type TreatmentDetailsDto = {
  id: number;
} & UpsertTreatmentDto;

export type TreatmentCategoryDto = {
  id: number;
  name: string;
};